_extras/sequence_alignment.md

```diff
@@ -0,0 +1,516 @@
+
+Since the container uses  `ENTRYPOINT` at this time we can only use it from outside the container *i.e.* adding `-it` trying 
+to make the container interactive would have no effect. We can provide the minimal mandatory input required 
+as the input and output file names with `-i` and `-o` which will pass directly to the `clustalo` program.
+`clustalo` expects data to be found in the `/data` directory created when we spawned the container.
+
+~~~
+$ docker run --rm --mount type=bind,source=${PWD},target=/data pegi3s/clustalomega -i /data/peptides.fasta -o /data/peptides_aligned.fasta
+$ cat peptides_aligned.fasta
+~~~
+{: .language-bash}
+~~~
+>GIP
+YAEGTFISDYSIAMDKIRQQDFVNWLLAQ----
+>GLP-1
+HAEGTFTSDVSSYLEGQAAKEFIAWLVKGRG--
+>GLP-2
+HADGSFSDEMNTILDNLAARDFINWLIQTKITD
+>glucagon
+HSQGTFTSDYSKYLDSRRAQDFVQWLMNT----
+~~~
+{: .output}
+
+The `-` symbols represent the gaps. Therefore it worked. 
+
+Note: running the same command will provide an error as `clustalo` will refuse to overwrite the existing `peptides_aligned.fasta` file. 
+This is a good thing! However, it is possible to override this safety by adding **`--force`** to the command. 
+
+`clustalo` can also output the alignment in other formats as detailed in the help under the optional modifier 
+`--outfmt`. We can try the `clustal` format for a more visual output by adding `--outfmt=clu`:
+
+~~~
+$ docker run --rm --mount type=bind,source=${PWD},target=/data pegi3s/clustalomega --outfmt=clu -i /data/peptides.fasta -o /data/peptides_aligned.clu
+$ cat peptides_aligned.clu
+~~~
+{: .language-bash}
+~~~
+CLUSTAL O(1.2.4) multiple sequence alignment
+
+
+GIP           YAEGTFISDYSIAMDKIRQQDFVNWLLAQ----
+GLP-1         HAEGTFTSDVSSYLEGQAAKEFIAWLVKGRG--
+GLP-2         HADGSFSDEMNTILDNLAARDFINWLIQTKITD
+glucagon      HSQGTFTSDYSKYLDSRRAQDFVQWLMNT----
+              :::*:* .: .  ::    ::*: **:      
+~~~
+{: .output}
+
+> ## Challenge: align more sequences.
+> How would you proceed to align the hemogloin and globin sequences contained within the file `hglob.fasta`? located in the `~/Desktop/docker-intro/peptides` you copied?
+> 
+> Would anything be different to align the 32 spike protein sequence from virus SARS_Cov-2 found in file `spike_32.fa` within the `~/Desktop/docker-intro/covid` directory?
+> 
+> > ## Solution
+> >
+> > Since we cannot run (for now) the container interactively we need to provide names for the input file and output result files. We can also select an alternate output format.
+> > ~~~
+> > $ docker run --rm --mount type=bind,source=${PWD},target=/data pegi3s/clustalomega --outfmt=clu -i /data/hglob.fasta -o /data/hglob_aligned.clu
+> > ~~~
+> > {: .language-bash}
+> > For the larger set in `spike_32.fa` we could add `-v` (verbose) to obtain more information from `clustalo` as the software is computing the alignment. 
+> > Nothing else needs to be fundamentally different. Just remember to add `--force` to run the same command again.
```

Comment:
Why do they need to force this?  Did they run it before at some point?

Reply:
If they try to run the same command again there will be an error, easily misunderstood. Adding `--force` is necessary, as written: "to run the same command again."